docs/book/src/developer/architecture/controllers/control-plane.md

```diff
@@ -50,6 +50,9 @@ The `ImplementationControlPlane` *must* rely on the existence of
 
 ### CRD contracts
 
+The CRD name must have the format `fmt.Sprintf("%s.%s", flect.Pluralize(strings.ToLower(Kind)), Group)`.
```

Comment:
Can we add in a non-code expression / example of what the name should look like? (and the same for the other examples below)

Reply:
The problem is that it's highly misleading, e.g. `metal3datas.infrastructure.cluster.x-k8s.io` is wrong, `metal3data.infrastructure.cluster.x-k8s.io` is correct. So I want to enforce this contract explicitly with this func as this is what it depends on in our code.

Any other plural forms that people come up with has a chance to be incorrect.

(Btw this will be 100% enforced once we drop the support for the invalid names, because CAPI will stop working anymore with invalid CRD names. So folks will notice the mistake very early when developing a new provider)

Reply:
I think that being explicit in documenting that plural must match what is returned by flect.Pluralize(kind) is relevant, let's not loose this detail

Reply:
Can we export this function in Cluster API with a link to a google playground to test it?

Reply:
I exposed a function and referenced it in the doc. 

I'm not sure what I'm doing wrong but I couldn't get playground to work:
https://go.dev/play/p/dsQkcJ-Q8p2
Leads to `timeout running go build`

Reply:
I saw bunch of similar reported issue upstream (https://github.com/golang/go/issues/44537, https://github.com/golang/go/issues/44822) which seems usual and even fix for incrasing compile timeout https://go-review.googlesource.com/c/playground/+/302771/ but looks like it has to be revised again 😀

Reply:
Also, @sbueringer as you pointed out if CAPI stops working with incorrect CRD names in the future and since metal3datas CRD is not used by core Cluster API, it should not create us (metal3 provider) down the road. But, just thinking out loud, I wonder if some exceptions (though it would create inconsistency in CRDs') can be made for such situations as we have now, like `data` basically using uncountable words in provider CRD names if needed?  https://pkg.go.dev/github.com/go-openapi/inflect#AddUncountable

Reply:
The problem is that if we want to make that exception we either have to define our own pluralize which diverges from the one from flect or we have to handle both possible names, which means matching for certain strings and then trying various CRD names.